### PR TITLE
Added Users to TableQueries

### PR DESCRIPTION
**Pull request by ayush-shah**
### Describe your changes :

Worked on adding users to tableQueries
#
### Type of change :

- [x] Improvement

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have performed a self-review of my own. 
- [x] I have tagged my reviewers below.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

@harshach @Sachin-chaurasiya @darth-coder00

### CODE REVIEW

File: ingestion/src/metadata/ingestion/source/sample_data.py [hunk 1]
```diff
@@ -339,10 +341,14 @@ def ingest_tables(self) -> Iterable[OMetaDatabaseAndTable]:
             ),
         )
         for table in self.tables["tables"]:
-            table_metadata = Table(**table)
-            table_and_db = OMetaDatabaseAndTable(table=table_metadata, database=db)
-            self.status.scanned("table", table_metadata.name.__root__)
-            yield table_and_db
+            resp = self.metadata.list_entities(entity=User, limit=5).json()
+            for sql_object in table["tableQueries"]:
+                id = json.loads(resp).get("entities")[random.choice(range(5))]["id"]
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/source/sample_data.py:346]: To make it easier to handle, the `list_entities` method in the API returns an `EntityList` object. If you want only specific `User`s, you can run:

```python
res = self.metadata.list_entities(entity=User, limit=5)
users = res.entities  # This is now a `List[User]`
```

In the API everything should be properly typed for easy building and retrieval. If some castings are missing we can open an issue to get them fixed

Thanks!

Comment (pmbrull) [reply ingestion/src/metadata/ingestion/source/sample_data.py:346]: Link to [EntityList](https://github.com/open-metadata/OpenMetadata/blob/main/ingestion/src/metadata/ingestion/ometa/ometa_api.py#L85)

**Comment by ayush-shah** [comment]
@pmbrull can you check if this is the right approach? I have modified the code according to the comments

**Review by pmbrull — APPROVED** [review]
Thanks for the changes @ayush-shah, looks great now 🚀

**Comment by pmbrull** [comment]
Also, do not hesitate to reach out if there are any doubts about the code or if you think something can be improved, always happy to discuss :)

**Comment by pmbrull** [comment]
Flagged the security hotspot as safe, as only affects sample data. No further action needed

**Pull request merged**